types/iobroker/iobroker-tests.ts

```diff
@@ -0,0 +1,5 @@
+// Help! what to test here?
```

Comment:
I'm having trouble understanding a couple things:
- are these types describing the ioBroker package or the ioBroker.js-controller package? If the latter, the name should be changed
- How would you expect to use the types in this package? I can't quite tell from the docs how you would use it, since it seems like the ioBroker.js-controller package doesn't actually export anything. If it's meant to just be a collection of types available to someone who would be writing to match some sort of adapter spec, then you should just use as many of the types you are exporting in this test file so you know if future changes cause any regressions. 

Basically, in this file, you just want to use the package as a user would.

Reply:
1.:
An adapter is basically a plugin that is executed by the `iobroker.js-controller` in a separate process. Currently it using a file like [this one](https://github.com/AlCalzone/ioBroker.tradfri/blob/master/src/lib/utils.ts) to access certain methods/classes necessary for the execution, including the `Adapter` class I declared in these definitions.
As part of an upcoming refactoring, these files are no longer included in every adapter, but will in a separate package `iobroker.adapter-core` - I guess this should be the correct name for the typings then? I referenced `iobroker.js-controller` in the comments because it currently determines how the declarations have to look like.

2.:
An example usage can be seen in https://github.com/AlCalzone/ioBroker.tradfri/blob/master/src/main.ts  
The user imports `iobroker.adapter-core` and then creates an adapter instance:
```ts
import utils from "iobroker.adapter-core";

let adapter: utils.adapter({
	name: "tradfri",
	ready: () => { /* ... */ },
	// ...
});
```
The instance `adapter` is of the type `ioBroker.Adapter` declared in line 926, which is the most important part of this declaration. It has several event handlers and methods to interact with `js-controller`.